repos/system_upgrade/el7toel8/actors/checksendmail/actor.py

```diff
@@ -0,0 +1,47 @@
+from leapp.actors import Actor
+from leapp.libraries.actor import library
+from leapp.models import InstalledRedHatSignedRPM, SendmailMigrationDecision, TcpWrappersFacts
+from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
+from leapp.reporting import Report
+from leapp.libraries.common.reporting import report_with_remediation, report_generic
+from leapp.libraries.common.tcpwrappersutils import config_applies_to_daemon
+
+
+class CheckSendmail(Actor):
+    """
+    Check if sendmail is installed, check whether configuration update is needed, inhibit upgrade if TCP wrappers are used.
+    """
+
+    name = 'check_sendmail'
+    consumes = (InstalledRedHatSignedRPM, TcpWrappersFacts,)
+    produces = (Report, SendmailMigrationDecision,)
+    tags = (ChecksPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        for fact in self.consume(InstalledRedHatSignedRPM):
+            for rpm in fact.items:
+                if rpm.name == 'sendmail':
+                    if config_applies_to_daemon(next(self.consume(TcpWrappersFacts)), 'sendmail'):
+                        report_with_remediation(
+                            title='TCP wrappers support removed in the next major version',
+                            summary='TCP wrappers are legacy host-based ACL (Access Control List) system '
+                                    'which has been removed in the next major version of RHEL.',
+                            remediation='Please migrate from TCP wrappers to some other access control mechanism and delete sendmail '
+                                    'from the /etc/hosts.[allow|deny].',
+                            severity='high',
+                            flags=['inhibitor'])
+                        return
+                    migrate_files = library.check_files_for_compressed_ipv6()
```

Comment:
Sorry if I'm not getting it: why are L34-L47 under the for loop?

Reply:
Which 'for'? From the line 21? It searches for the sendmail package, if found, it checks whether migration is needed, then the loop is exited by 'return' on line 47.

Reply:
Still not getting why this logic can’t be at the same level of nesting as for loop at line 21.

Reply:
The config should be migrated only if the sendmail package is installed. I could restruct the code from the lines 34 up and unindent it to level of line 21, but I would have to add more control logic / another variable to note if sendmail package was found or not. Is it worth the complication?

I.e. it should work the following way:
1. find out whether sendmail package is installed, if not do nothing
2. check if tcp_wrappers are used, if yes, set inhibitor, do not continue with the migration
3. collect files for migration and check whether migration is needed

Step 1 is written as the for loop in question for slight speed optimization, so when the first package named 'sendmail' is found, steps 2 - 3 are run and then it exits so the loop doesn't continue.

Reply:
My fault, I somehow missed return at L47 and raised a red flag in my head. 
Dismissing this concern.